### PR TITLE
Added support for hideki wind sensor.

### PR DESCRIPTION
**Pull request by Matthies**

### CODE REVIEW

**Comment by merbanan** [comment]
Looks good, but you need to write a commit log that says what the commit does. Can you update that so I can merge the changes.

**Comment by Matthies** [comment]
Probably not possible to add message to older commit so I added another commit with the message.

**Comment by zuckschwerdt** [comment]
@Matthies with git there'll always be a – nay, at least one – way:) Look for "squashing". I this case I'd
```shell
git reset --soft HEAD~2
git add .
git commit --amend -m "commit message"
git push -f
```

**Comment by ludwich66** [comment]
Hi i had integrated the changes manually, it works proper on my 
HIDEKI WEATHERstation
my, Conrad TE831  Weather Station // https://www.conrad.de/de/funk-wetterstation-te831x-646329.html
CRESTA BAR 818

There is an complet Protocol 
http://members.upc.nl/m.beukelaar/Crestaprotocol.pdf

If if the Protokoll complete in RTL_433, i search and install the Rainsensor for an test ;-)

LOG with changes from Matthias
Please merge it :-)

2016-10-16 10:44:43 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.5 C
	Wind Strength:	 0.00 km/h
	Direction:	 247.5 °
2016-10-16 10:44:43 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.5 C
	Wind Strength:	 0.00 km/h
	Direction:	 247.5 °
2016-10-16 10:44:44 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.5 C
	Wind Strength:	 0.00 km/h
	Direction:	 247.5 °
2016-10-16 10:45:02 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.1 C
	Humidity:	 79 %
2016-10-16 10:45:03 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.1 C
	Humidity:	 79 %
2016-10-16 10:45:03 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.1 C
	Humidity:	 79 %
2016-10-16 10:45:16 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.5 C
	Wind Strength:	 0.00 km/h
	Direction:	 292.5 °
2016-10-16 10:45:45 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.1 C
	Humidity:	 78 %
2016-10-16 10:45:46 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.1 C
	Humidity:	 78 %
2016-10-16 10:45:46 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.1 C
	Humidity:	 78 %
2016-10-16 10:45:49 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.6 C
	Wind Strength:	 0.00 km/h
	Direction:	 270.0 °
2016-10-16 10:45:49 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.6 C
	Wind Strength:	 0.00 km/h
	Direction:	 270.0 °
2016-10-16 10:45:50 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.6 C
	Wind Strength:	 0.00 km/h
	Direction:	 270.0 °
2016-10-16 10:46:22 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.6 C
	Wind Strength:	 0.00 km/h
	Direction:	 45.0 °
2016-10-16 10:46:22 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.6 C
	Wind Strength:	 0.00 km/h
	Direction:	 45.0 °
2016-10-16 10:46:22 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.6 C
	Wind Strength:	 0.00 km/h
	Direction:	 45.0 °
2016-10-16 10:46:28 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:46:29 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:46:29 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:46:55 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.6 C
	Wind Strength:	 0.00 km/h
	Direction:	 67.5 °
2016-10-16 10:46:55 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.6 C
	Wind Strength:	 0.00 km/h
	Direction:	 67.5 °
2016-10-16 10:47:11 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:47:12 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:47:12 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:47:28 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.6 C
	Wind Strength:	 0.00 km/h
	Direction:	 247.5 °
2016-10-16 10:47:28 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.6 C
	Wind Strength:	 0.00 km/h
	Direction:	 247.5 °
2016-10-16 10:47:28 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.6 C
	Wind Strength:	 0.00 km/h
	Direction:	 247.5 °
2016-10-16 10:47:54 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:47:55 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:47:55 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:48:02 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.7 C
	Wind Strength:	 0.00 km/h
	Direction:	 247.5 °
2016-10-16 10:48:02 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.7 C
	Wind Strength:	 0.00 km/h
	Direction:	 247.5 °
2016-10-16 10:48:02 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.7 C
	Wind Strength:	 0.00 km/h
	Direction:	 247.5 °
2016-10-16 10:48:35 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.7 C
	Wind Strength:	 0.00 km/h
	Direction:	 292.5 °
2016-10-16 10:48:35 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.7 C
	Wind Strength:	 0.00 km/h
	Direction:	 292.5 °
2016-10-16 10:48:35 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.7 C
	Wind Strength:	 0.00 km/h
	Direction:	 292.5 °
2016-10-16 10:48:37 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:48:38 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:48:38 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:49:08 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.8 C
	Wind Strength:	 0.00 km/h
	Direction:	 315.0 °
2016-10-16 10:49:08 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.8 C
	Wind Strength:	 0.00 km/h
	Direction:	 315.0 °
2016-10-16 10:49:08 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.8 C
	Wind Strength:	 0.00 km/h
	Direction:	 315.0 °
2016-10-16 10:49:20 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:49:21 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:49:21 :	HIDEKI TS04 sensor
	Rolling Code:	 9
	Channel:	 1
	Battery:	 OK
	Temperature:	 15.2 C
	Humidity:	 78 %
2016-10-16 10:49:40 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.8 C
	Wind Strength:	 0.00 km/h
	Direction:	 90.0 °
2016-10-16 10:49:41 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.8 C
	Wind Strength:	 0.00 km/h
	Direction:	 90.0 °
2016-10-16 10:49:41 :	HIDEKI Wind sensor
	Rolling Code:	 2
	Channel:	 4
	Battery:	 OK
	Temperature:	 15.8 C
	Wind Strength:	 0.00 km/h
	Direction:	 90.0 °

Yes there is no wind today !
The data are valide

Ludwich

**Comment by merbanan** [comment]
@ludwich66  can you send some signal samples to the tests repository ? I will then fix and merge the code. There are changes in the demodulator I don't know if they are correct.

**Comment by ludwich66** [comment]
Hi, Benjamin,
 i used a actually rtl-433 snapshot from today, is the demodulator change newer?
i hope its not complicated with the Signal sales - within is not my friend :-)
ludwich

--> do you have seen the link to the protocol?
Is there a way to add the rain sensor?

**Comment by merbanan** [comment]
@ludwich66 I don't really understand what you mean. But I wont merge this code unless I have some signals to test against. So can you send some test signals ?

https://github.com/merbanan/rtl_433#supporting-additional-devices-and-test-data

**Comment by ludwich66** [comment]
Sorry for the auto uncorection :-)
i have sent samples to rtf_433 test
ludwich

**Comment by ludwich66** [comment]
where is the problem? i receive with rtl_433 -a -t only the Hideki ts 04 data. there ar no Hideki Wind Sensor data? in mode rtl_433 -R  21 (with patch) i do receive 3x Hideki ts04 followd from 3x Hideki Wind Sensor ??
ludwich

**Comment by merbanan** [comment]
The patch changes src/pulse_demod.c, I need to make sure that it is needed and that the change doesn't break any other decoders. So the easiest way to do that is to investigate the samples. Which I have none until you send the files to the rtl_433_tests repository. Regarding rain support there just is not code for that. With the specs you linked it should be easy to add but once again I have no good sample files to work with. So to make myself clear, no sample signal files -> no support.

**Comment by ludwich66** [comment]
did you see the comment before? why do i not receive the Wind Sensor in -a -t mod?
is it better to use the unmodified RTL-433 version ?
any ideas?
Next samples follow today evening.
had a nice day
ludwich

**Comment by rct** [comment]
The behavior is when running with the -a analyzer decoding is disabled.

The typical development cycle is

1. record new undecided signals with -a -t
2. Replay with -r gfileNNN.data while you work on your decoder.
3. Test changes against your known stable recordings as well as live.

**Comment by ludwich66** [comment]
Hi Benjamin,
?? now with a shorter distance between Sensor and Computer it was possible to select new Testdata.
The first Part are from TS04 the actually data are only the hideki Windsensor.
If this ok for you?
ludwich

**Comment by merbanan** [comment]
Send what you get, that is all you can do.

**Comment by Matthies** [comment]
Some data from my hideki windsensor.
[hideki-wind-sensor-data.zip](https://github.com/merbanan/rtl_433/files/545660/hideki-wind-sensor-data.zip)

**Comment by ludwich66** [comment]
What is needed before the wind sensor is implemented?
ludwich

**Comment by Matthies** [comment]
I have recorded some data from the Hideki wind sensor recorded with my 
patched rtl_433 -a -t

Attached to this mail.

Regards, Andreas


Am 17.10.2016 um 19:51 schrieb Benjamin Larsson:

**Comment by ludwich66** [comment]
HIDEKI Rainsensor 
i have added 3 test files gfile061.data gfile072.data gfile082.data
The encoding is wrong
1: 234.9mm 
1 to 2 = 9.2mm
2: 244.1mm
2 to 3 = 7.7mm
3: 251.8mm
Terminal with wrong encoding
est mode active. Reading samples from file: gfile082.data
Input format: uint8
@0.000000s :	HIDEKI TS04 sensor
	Rolling Code:	 0
	Channel:	 4
	Battery:	 LOW
	Temperature:	 -43.2 C
	Humidity:	 66 %
@0.524288s :	HIDEKI TS04 sensor
	Rolling Code:	 0
	Channel:	 4
	Battery:	 LOW
	Temperature:	 -43.2 C
	Humidity:	 66 %
@0.524288s :	HIDEKI TS04 sensor
	Rolling Code:	 0
	Channel:	 4
	Battery:	 LOW
	Temperature:	 -43.2 C
	Humidity:	 66 %

do the next step ;-)
step by step

ludwich

**Comment by ludwich66** [comment]
Is it possible for you to add the Rainsensor too?
i walked to my attic and found the rainsensor, its time to play :-)

ludwich

**Comment by Matthies** [comment]
Where can I find the three sample files from your rain sensor?

**Comment by ludwich66** [comment]
Ich komme mit Git-Hub einfach nicht klar :-(
Hier nochmals als Anlage.

I dont understand Git-Hub - creasy System :-(
The Rainsensorfiles as attachment

ludwich

**Comment by Matthies** [comment]
Nope. Still no attachement in your mail.

File: src/devices/hideki.c [hunk 1]
```diff
@@ -37,31 +47,54 @@ static int hideki_ts04_callback(bitbuffer_t *bitbuffer) {
         // check parity
         uint8_t parity = ((b[i+offset+1] >> (7 - i%8)) ^ 0xFF) & 0x01;
         if(parity != byteParity(packet[i]))
-            return 0;
```
Comment (merbanan) [inline src/devices/hideki.c]: Why do you change this ?

**Comment by ludwich66** [comment]
Hi Benjamin,

I can not say anything about this change.
At the moment

Hideki TS04 (Temp/Humidity)
Hideki Wind (Windspeed, Winddirection, Temp/WindchiIl)
Hideki Rain (Rain shovel)

State of the Rainsensor —There is now a incorrect Battary Level, but it is very difficult to simulate. Yesterday i received  gfile095.date, i hope it is with Bat Lowlevel. i lost the connection between Wethterstation before …)

With the RTL USB stick I have a much better reception than with the weather station -da, I sometimes have to clean the antenna ;-)

Thanks ludwich

File: src/devices/hideki.c (hunk 1 above)
Comment (Matthies) [inline src/devices/hideki.c]: To handle the different data length of the different types of sensors. But I agree that the code should return immediately if parity error occurs and the datalength differs from every type of sensor. Will fix in next pull request.

**Comment by ludwich66** [comment]
That looks good.

did you see the gfile095.data? i hope there is a Low Bat Flag set.

ludwich

**Comment by Matthies** [comment]
Yes, I saw but not much time these days. Maybe this evening, maybe at the weekend.


Gesendet: Mittwoch, 16. November 2016 um 10:52 Uhr


Von: ludwich66 <notifications@github.com>
An: merbanan/rtl_433 <rtl_433@noreply.github.com>
Cc: Matthies <a.matthies@online.de>, Mention <mention@noreply.github.com>
Betreff: Re: [merbanan/rtl_433] Added support for hideki wind sensor. (#384)

did you see the gfile095.data? i hope there is a Low Bat Flag set.

**Comment by ludwich66** [comment]
Thanks,

As you know I have my dear need with Github

:-(

ludwich

**Pull request closed without merging**